Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,198 @@
+
+During .NET class registration, the shim is identified as the in-proc server for the class. Additional metadata is inserted into the registry to indicate what .NET assembly to load and what type to activate. For example, in addition to the typical [in-proc server](https://docs.microsoft.com/en-us/windows/desktop/com/inprocserver32) registry values the following values are added to the registry for the `TypeLoadException` class.
+
+```
+"Assembly"="mscorlib, Version=1.0.5000.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
+"Class"="System.TypeLoadException"
+"RuntimeVersion"="v1.1.4322"
+```
+
+The above registration is typically done with the [`RegAsm.exe`](https://docs.microsoft.com/en-us/dotnet/framework/tools/regasm-exe-assembly-registration-tool) tool. Alternatively, registry scripts can be generated by `RegAsm.exe`.
+
+### .NET Core Class COM Activation
+
+In .NET Core, our intent will be to avoid a system wide shim library. This decision may add additional cost for deployment scenarios, but will reduce servicing and engineering costs by making deployment more explicit and less magic.
+
+The current .NET Core hosting solutions are described in detail at [Documentation/design-docs/host-components.md](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-components.md). Along with the existing hosts an additional activation host library will be added. This library (henceforth identified as 'shim') will export the required functions for COM class activation and registration and act in a way similar to `mscoree.dll`.
+
+>[`HRESULT DllGetClassObject(REFCLSID rclsid, REFIID riid, LPVOID *ppv);`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-dllgetclassobject)
+
+>[`HRESULT DllCanUnloadNow();`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-dllcanunloadnow)
+
+>[`HRESULT DllRegisterServer();`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682162(v=vs.85).aspx)
+
+>[`HRESULT DllUnregisterServer();`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms691457(v=vs.85).aspx)
+
+When `DllGetClassObject()` is called in a COM activation scenario, the following will occur:
+
+1) Determine additional registration information needed for activation.
+    * The shim will check for an embedded manifest. If the shim does not contain an embedded manifest, the shim will check if a file with the `<host_library_name>.clsidmap` naming format exists adjacent to it.
```

Comment:
Do we expect the shim going to be renamed to contain your project name by the tooling? 

I think it should get renamed to make it clear it was customized for your project.

Reply:
> Do we expect the shim going to be renamed to contain your project name by the tooling?

In the default case, yes. There should be an option to not rename, but that is definitely not the default. In this particular document I am focusing on CLR mechanics/contracts, not the specific UX around the dev experience. The renaming of the host and how it should be used in a secure manner are being left to the official CLI documentation and/or examples.